File: presentation/src/main/java/com/github/anrimian/simplemusicplayer/di/app/components/AppComponent.java
package com.github.anrimian.simplemusicplayer.di.app.components;



import com.github.anrimian.simplemusicplayer.di.app.modules.AppModule;

import javax.inject.Singleton;

import dagger.Component;

/**
 * Created on 11.02.2017.
 */

@Singleton
@Component(modules = { AppModule.class })
public interface AppComponent {


}

File: app-wear/src/main/java/com/github/anrimian/musicplayer/wear/di/app/AppComponent.kt
package com.github.anrimian.musicplayer.wear.di.app

import com.github.anrimian.musicplayer.di.app.SchedulerModule
import com.github.anrimian.musicplayer.wear.domain.WearStateInteractor
import com.github.anrimian.musicplayer.wear.domain.queue.PlayQueueInteractor
import com.github.anrimian.musicplayer.wear.ui.MainPresenter
import dagger.Subcomponent
import javax.inject.Singleton

@Singleton
@Subcomponent(modules = [
    AppModule::class,
    SchedulerModule::class
])
interface AppComponent {

    fun mainPresenter(): MainPresenter

    fun wearStateInteractor(): WearStateInteractor
    fun playQueueInteractor(): PlayQueueInteractor

}